websites/P/ProBot/presence.ts

```diff
@@ -9,34 +9,165 @@ presence.on("UpdateData", () => {
+		presenceData.details = "Managing their";
+		presenceData.state = "Profile Badges";
 	} else if (document.location.pathname.includes("/xp")) {
-		presenceData.details = "Viewing a page:";
-		presenceData.state = "Top 100 by XP";
+		presenceData.details = "Viewing top 100 people";
+		presenceData.state = "with the most XP";
+		presenceData.buttons = [
+			{
+				label: "Visit the page",
+				url: document.URL
+			}
+		];
 	} else if (document.location.pathname.includes("/credits")) {
-		presenceData.details = "Viewing a page:";
-		presenceData.state = "Richest 100 Billionaires";
+		presenceData.details = "Viewing top 100 people";
+		presenceData.state = "with the most Credits";
+		presenceData.buttons = [
+			{
+				label: "Visit the page",
+				url: document.URL
+			}
+		];
+	} else if (document.location.pathname.includes("/daily")) {
+		presenceData.details = "Claiming their daily Credits";
+		presenceData.buttons = [
+			{
+				label: "Claim your daily reward!",
+				url: document.URL
+			}
+		];
 	} else if (document.location.pathname.includes("/transactions")) {
-		presenceData.details = "Viewing a page:";
-		presenceData.state = "Credits Transactions";
+		presenceData.details = "Viewing their";
+		presenceData.state = "Credits Transactions History";
+	} else if (document.location.pathname.includes("terms-of-use")) {
+		presenceData.details = "Reading the Terms Of Use";
+		presenceData.buttons = [
+			{
+				label: "Read the page",
+				url: document.URL
+			}
+		];
+	} else if (document.location.pathname.includes("privacy-policy")) {
+		presenceData.details = "Reading the Privacy Policy page";
+		presenceData.buttons = [
+			{
+				label: "Read the page",
+				url: document.URL
+			}
+		];
+	} else if (document.location.pathname.includes("refund-policy")) {
+		presenceData.details = "Reading the Refund Policy page";
+		presenceData.buttons = [
+			{
+				label: "Read the page",
+				url: document.URL
+			}
+		];
+	} else if (document.location.pathname.includes("/server/")) {
+		presenceData.smallImageKey = "probot";
+		presenceData.largeImageKey = (
+			document.querySelector(
+				"#sidebar_sidebar__avatar__QDGRP"
+			) as HTMLImageElement
+		).src;
+		presenceData.smallImageText = document.querySelector(
+			"#sidebar_sidebar__server-info__03ViT > h4"
+		).textContent;
+		if (document.location.pathname.includes("/settings"))
+			presenceData.details = "Changing the Server Settings";
+		else if (document.location.pathname.includes("embed"))
+			presenceData.details = "Managing the Embed Messages";
+		else if (document.location.pathname.includes("panel_logs"))
+			presenceData.details = "Viewing Control Panel Logs";
+		else if (document.location.pathname.includes("mod_actions"))
+			presenceData.details = "Viewing Moderation Actions";
+		else if (document.location.pathname.includes("premium")) {
+			if (document.location.pathname.includes("/manage"))
+				presenceData.details = "Managing their Premium";
```

Comment:
why are you managing your premium again here?

Reply:
@ImRodry because it's for the server specific premium settings (tier 1)

Reply:
no it isn't, the if you had above already covers this case